src/System.Net.Security/src/System/Net/Security/SslState.cs

```diff
@@ -1346,25 +1364,31 @@ internal void FinishWrite()
 
             lock (this)
             {
-                object obj = _queuedWriteStateRequest;
-                if (obj == null)
-                {
-                    // A repeated call.
-                    return;
-                }
+                HandleQueuedCallback(ref _queuedWriteStateRequest);
+            }
+        }
 
-                _queuedWriteStateRequest = null;
-                if (obj is LazyAsyncResult)
-                {
-                    // Sync handshake is waiting on other thread.
-                    ((LazyAsyncResult)obj).InvokeCallback();
-                }
-                else
-                {
-                    // Async handshake is pending, start it on other thread.
-                    // Consider: we could start it in on this thread but that will delay THIS write completion
+        private void HandleQueuedCallback(ref object queuedStateRequest)
+        {
+            object obj = queuedStateRequest;
+            queuedStateRequest = null;
+            switch (obj)
+            {
+                case null:
```

Comment:
How common is null? If it's common, you might save a write through the ref by checking this first and then only writing null to queuedStateRequest after that if it's not null.

Reply:
Null is super common so I put in early exit on null